darts/models/forecasting/regression_model.py

```diff
@@ -578,7 +599,10 @@ def predict(
         series_matrix = None
         if "target" in self.lags:
             series_matrix = np.stack(
-                [ts[self.lags["target"][0] :].values(copy=False) for ts in series]
+                [
+                    ts[self.lags["target"][0] - shift :].values(copy=False)
```

Comment:
(nitpicking) I think it would be better to first extract the Numpy array and then index it, rather than doing the indexing on the TimeSeries first. That's because indexing a TimeSeries creates a new TimeSeries, which can be relatively expensive.

Reply:
I see, I did not want to change too much the existing logic, but I can correct that